pkg/operator/sync.go

```diff
@@ -587,28 +577,46 @@ func (optr *Operator) syncRequiredMachineConfigPools(_ *renderConfig) error {
 				return false, nil
 			}
 		}
-		pools, err := optr.mcpLister.List(sel)
+
+		pools, err := optr.mcpLister.List(labels.Everything())
 		if err != nil {
 			lastErr = err
 			return false, nil
 		}
 		for _, pool := range pools {
-			if err := isMachineConfigPoolConfigurationValid(pool, version.Hash, optr.mcLister.Get); err != nil {
-				lastErr = fmt.Errorf("pool %s has not progressed to latest configuration: %v, retrying", pool.Name, err)
+			degraded := isPoolStatusConditionTrue(pool, mcfgv1.MachineConfigPoolDegraded)
+			if degraded {
+				lastErr = fmt.Errorf("error pool %s is not ready, retrying. Status: (pool degraded: %v total: %d, ready %d, updated: %d, unavailable: %d)", pool.Name, degraded, pool.Status.MachineCount, pool.Status.ReadyMachineCount, pool.Status.UpdatedMachineCount, pool.Status.UnavailableMachineCount)
+				glog.Errorf("Error syncing Required MachineConfigPools: %q", lastErr)
+				syncerr := optr.syncUpgradeableStatus()
+				if syncerr != nil {
+					glog.Errorf("Error syncingUpgradeableStatus: %q", syncerr)
+				}
 				return false, nil
 			}
-			degraded := isPoolStatusConditionTrue(pool, mcfgv1.MachineConfigPoolDegraded)
-			if pool.Generation <= pool.Status.ObservedGeneration &&
-				isPoolStatusConditionTrue(pool, mcfgv1.MachineConfigPoolUpdated) &&
-				!degraded {
-				continue
+
+			_, hasRequiredPoolLabel := pool.Labels[requiredForUpgradeMachineConfigPoolLabelKey]
+
+			if hasRequiredPoolLabel {
+				if err := isMachineConfigPoolConfigurationValid(pool, version.Hash, optr.mcLister.Get); err != nil {
+					lastErr = fmt.Errorf("pool %s has not progressed to latest configuration: %v, retrying", pool.Name, err)
+					return false, nil
+				}
+
+				if pool.Generation <= pool.Status.ObservedGeneration &&
+					isPoolStatusConditionTrue(pool, mcfgv1.MachineConfigPoolUpdated) {
+					continue
+				}
 			}
-			lastErr = fmt.Errorf("error pool %s is not ready, retrying. Status: (pool degraded: %v total: %d, ready %d, updated: %d, unavailable: %d)", pool.Name, degraded, pool.Status.MachineCount, pool.Status.ReadyMachineCount, pool.Status.UpdatedMachineCount, pool.Status.UnavailableMachineCount)
-			return false, nil
+		}
+		syncstatuserr := optr.syncUpgradeableStatus()
```

Comment:
is this extra as we're syncing upgradeable in the loop above? (maybe my eyes are tricking me lol)

Reply:
no we're only syncing if it's degraded above (it syncs then returns false), this needs to be here to pick up with it's not degraded, without it here it won't toggle back when it fixed.

Reply:
won't it? :thinking: this function returns to `syncAll` which in turns calls `syncUpgradeableStatus` again right?

Reply:
postponed :) not harmful